include/xbps_api_impl.h

```diff
@@ -135,14 +155,15 @@ bool HIDDEN xbps_transaction_store(struct xbps_handle *, xbps_array_t, xbps_dict
 int HIDDEN xbps_transaction_init(struct xbps_handle *);
 int HIDDEN xbps_transaction_files(struct xbps_handle *,
 		xbps_object_iterator_t);
+void xbps_transaction_files_free(void);
 int HIDDEN xbps_transaction_fetch(struct xbps_handle *,
 		xbps_object_iterator_t);
 int HIDDEN xbps_transaction_pkg_deps(struct xbps_handle *, xbps_array_t, xbps_dictionary_t);
 
+struct transaction_file *xbps_transaction_file_new(struct xbps_handle *, const char *);
+
```

Comment:
The naming is a bit confusing, in that `xbps_transaction_file_new` was added, but no `xbps_transaction_file_free`, and `xbps_transaction_files_free` was added, without a corresponding `_new`, and doesn't seem to be called at all.

Reply:
Yes the _new does not actually create anything, it gives you a reference to a file struct that already exists from collecting all files in the transaction, "_new" because it returns the new file as opposed to the old file.

Not sure how this should be done exactly, something like `xbps_transaction_get_file` with either _new or _old or a parameter to choose which one you want.

Reply:
`xbps_transaction_files_free` deletes the whole hashtable/tree, this should be called from somewhere after the transaction is done, maybe from `xbps_transaction_commit`.

Reply:
> Not sure how this should be done exactly, something like xbps_transaction_get_file with either _new or _old or a parameter to choose which one you want.

I think the parameter helps avoid duplicated logic, so I'm in favor of that.